### ISSUE TITLE
pdf conversion callback never fired

### ISSUE THREAD

**Issue by jessekinsman** [hacksparrow/node-easyimage#54]
If I do a pdf conversion on a multi-page doc, the callback is never fired. 

I am guessing this is because the promise is only fulfilled once?

Here is the code I am using:
<code><pre>
    easyimg.convert({src: multiPagePDF, dst: image.jpg},
    function(err, stdout, stderr) {
        if (err) throw err;
        console.log("done");
    });
</pre>
</code>

**Comment by dolohow** [comment]
Just look at the code in examples.
First of all you have no width or height in your options object.
Also easyimage is promise oriented. You should use .then as stated in README.

**Comment by jessekinsman** [comment]
I appreciate you responding, but if you look at the readme.md at the time I was working on this problem, using the .then was not part of the examples
https://github.com/hacksparrow/node-easyimage/blob/79c4a2a6ff533f9eadcc6c088b71f4ccd52a734d/README.md

I have since implemented the .then but still have been unable to get it to work due to this error
``[Error: identify: unable to open image `test/pages/page.jpg': No such file or directory @ error/blob.c/OpenBlob/2695.
]``

It is strange that it throws this error since it actually created the images. Not sure why it is calling the identify method on the images after it has already created them. I am pretty sure this has something to do with the multi-page pdf and it not finding the actual file since a page number has been appended to the destination (i.e. page.jpg turns into page-01.jpg).

And can you please show me the code in the examples that converts a multi-page pdf to images?

For those who are running into the same problem, I have gotten this to work but using the exec command
```
easyimg.exec('convert -density 150 ' + sourcePDF + ' ' + destination).then(
			function(){
				console.log("done with pdf conversion");
			},
			function(err) {
				console.log("pdf conversion error");
				if (err) throw err;
			}
		);
```
Hope that helps!

**Comment by tylerevans** [comment]
Was this ever resolved?

**Comment by hacksparrow** [comment]
@tylerevans no. Feel free to open a PR.

**Comment by jessekinsman** [comment]
I don't think so. I haven't tested recently. I think the openblob error may have something to do with mogrify.

**Comment by mrkmg** [comment]
@tylerevans is this still an issue?

**Comment by mrkmg** [comment]
This should be fixed as of #95
